include/envoy/network/filter.h

```diff
@@ -10,6 +10,8 @@ namespace Network {
 
 class Connection;
 class ConnectionSocket;
+class TransportSocket;
+typedef std::unique_ptr<TransportSocket> TransportSocketPtr;
```

Comment:
Just include transport socket here? If you use unique_ptr, this is going to force files including this file to include transport_socket as well.

Reply:
Unfortunately, including `transport_socket.h` here introduces circular dependency.

Reply:
I think you can change `transport_socket.h` a bit to avoid circular dependency by forward declaring `Network::Connection` there. Since it only use reference to it (which doesn't force includes).